source/interface/directoryContents.js

```diff
@@ -27,7 +27,13 @@ function getDirectoryContents(remotePath, options) {
         })
         .then(parseXML)
         .then(result => getDirectoryFiles(result, options.remotePath, remotePath, options.details))
-        .then(files => processResponsePayload(response, files, options.details));
+        .then(files => processResponsePayload(response, files, options.details))
+        .then(files => {
+            if (options.glob && options.glob.pattern) {
+                return processGlobFilter(files, options.glob);
+            }
+            return files;
```

Comment:
This could perhaps be made a tad simpler, something like `.then(files => options.glob && options.glob.pattern ? processGlobFilter(files, options.glob) : files)`